### PR TITLE
#2983 added keptn spec version to events sent by shipyard-controller

### LINKED ISSUE #2983: Add keptn/spec version to metadata of Keptn CloudEvents
CloudEvents must contain the keptn/spec version implemented by the shipyard-controller. The shipyard-controller needs to be configured with the keptn/spec version, e.g., in an environment variable the version is set. The shipyard-controller then sends out the events with the keptn/spec version. 

_Use case requirement_: Someone who is developing an integration knows exactly the keptn/spec version that is processed.

**Example:** 
```
{
  "datacontenttype": "application/json",
  "id": "c4d3a334-6cb9-4e8c-a372-7e0b45942f53",
  "source": "source-service",
  "specversion": "1.0",
  "keptnspecversion": "0.2.0-alpha",
  "type": "sh.keptn.event.approval.triggered",
  "data": {
    "project": "sockshop",
    "stage": "dev",
    "service": "carts",
    "labels": {
      "label-key": "label-value"
    },
    "status": "succeeded",
    "result": "pass",
    "message": "a message"
  }
}
```

**Tasks:**
* Add the property `keptnspecversion` to the meta-data of a CloudEvent in keptn/spec. 
* Add an automation step that takes the last released keptn/spec version and defines it for the shipyard-controller
* The shipyard-controller sends out `triggered` events containing the `keptnspecversion`
* go-utils functions that create the started/finished events based on the triggered event consider this new property; this is related to: https://github.com/keptn/keptn/issues/2913
* Out-of-scope: The API does not validate whether a started/triggered contains the keptnspecversion.

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #2983
Signed-off-by: Florian Bacher <florian.bacher@dynatrace.com>

### CODE REVIEW

File: mongodb-datastore/deploy/mongodb-datastore.yaml [hunk 1]
```diff
@@ -30,16 +30,16 @@ spec:
       serviceAccountName: keptn-default
       containers:
       - name: mongodb-datastore
-        image: keptn/mongodb-datastore:latest
+        image: keptn/mongodb-datastore:0.8.0-dev
```
Comment (warber) [inline mongodb-datastore/deploy/mongodb-datastore.yaml:33]: why the hardcoded version here?

Comment (bacherfl) [reply mongodb-datastore/deploy/mongodb-datastore.yaml]: was changed by accident; is fixed now

File: mongodb-datastore/deploy/mongodb-datastore.yaml [hunk 2]
```diff
@@ -35,11 +35,11 @@ spec:
         - containerPort: 8080
         resources:
           requests:
-            memory: "128Mi"
+            memory: "256Mi"
             cpu: "100m"
           limits:
-            memory: "256Mi"
-            cpu: "1000m"
+            memory: "512Mi"
```
Comment (warber) [inline mongodb-datastore/deploy/mongodb-datastore.yaml:41]: does the mongodb datastore really require more resources or is this a leftover from debugging?

Comment (bacherfl) [reply mongodb-datastore/deploy/mongodb-datastore.yaml:41]: the in-cluster debugging was not working properly in some cases - increasing the memory limit helped in that case

**Review by warber — APPROVED** [review]

**Pull request merged**